run0/src/container/command.rs

```diff
@@ -0,0 +1,42 @@
+use oci_spec::runtime::Process;
+
+const UNSHARE_PROGRAM_ENTRYPOINT: &str = "/bin/sh";
+const DEFAULT_ARG0: &str = "sh";
+
+/// Implementation of the container arguments.
+#[derive(Default)]
+pub struct Command {
+    arg0: String,
+    args: Vec<String>,
+}
+
+impl Command {
+    /// Get arguments.
+    /// It converts the `Self` into a `Command` for `unshare` API compatibility.
+    pub fn get(&self) -> unshare::Command {
+        let mut command = unshare::Command::new(UNSHARE_PROGRAM_ENTRYPOINT);
+        command.arg0(&self.arg0).args(&self.args);
+
+        command
+    }
+}
+
+impl From<&Option<Process>> for Command {
+    fn from(process: &Option<Process>) -> Self {
+        let mut args = Vec::<String>::new();
+        if let Some(process) = process {
+            if let Some(arguments) = process.args() {
+                for arg in arguments {
+                    args.push(arg.to_string());
+                }
+            }
+        }
+
+        let arg0 = match args.len() {
+            0 => String::from(DEFAULT_ARG0),
+            _ => args.remove(0),
+        };
+
```

Comment:
Now that you start from a default Command, you would just need to check if `arguments.is_empty()` from the previous loop and do a `command.arg0 = command.args.remove(0);` if it is not.